apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -2293,19 +2214,225 @@ func getOperatorOwner(r *ReconcileAPI) ([]metav1.OwnerReference, error) {
+
+	//get all the securities defined in swagger
+	var securityMap = make(map[string][]string)
+	var securityDef path
+	//get API level security
+	apiLevelSecurity, isDefined := swagger.Extensions[securityExtension]
+	var APILevelSecurity []map[string][]string
+	if isDefined {
+		log.Info("API level security is defined")
+		rawmsg := apiLevelSecurity.(json.RawMessage)
+		errsec := json.Unmarshal(rawmsg, &APILevelSecurity)
+		if errsec != nil {
+			log.Error(errsec, "error unmarshaling API level security ")
+			return securityMap, isDefined, len(securityDef.Security), errsec
+		}
+		for _, value := range APILevelSecurity {
+			for secName, val := range value {
+				securityMap[secName] = val
+			}
+		}
+	} else {
+		log.Info("API Level security is not defined")
+	}
+	//get resource level security
+	resLevelSecurity, resSecIsDefined := swagger.Extensions[pathsExtension]
+	var resSecurityMap map[string]map[string]path
+
+	if resSecIsDefined {
+		rawmsg := resLevelSecurity.(json.RawMessage)
+		errrSec := json.Unmarshal(rawmsg, &resSecurityMap)
+		if errrSec != nil {
+			log.Error(errrSec, "error unmarshall into resource level security")
+			return securityMap, isDefined, len(securityDef.Security), errrSec
+		}
+		for _, path := range resSecurityMap {
+			for _, sec := range path {
+				securityDef = sec
+
+			}
+		}
+	}
+	if len(securityDef.Security) > 0 {
+		log.Info("Resource level security is defined")
+		for _, obj := range resSecurityMap {
+			for _, obj := range obj {
+				for _, value := range obj.Security {
+					for secName, val := range value {
+						securityMap[secName] = val
+					}
+				}
+			}
+		}
+	} else {
+		log.Info("Resource level security is not defiend")
+	}
+	return securityMap, isDefined, len(securityDef.Security), nil
+}
+
+func handleSecurity(r *ReconcileAPI, securityMap map[string][]string, userNameSpace string, instance *wso2v1alpha1.API, secSchemeDefined bool, certList map[string]string, jobVolumeMount []corev1.VolumeMount, jobVolume []corev1.Volume) (map[string]securitySchemeStruct, bool, map[string]string, []corev1.VolumeMount, []corev1.Volume, []SecurityTypeJWT, error) {
+
+	var alias string
+	//keep to track the existence of certificates
+	var existSecCert bool
+
+	var securityDefinition = make(map[string]securitySchemeStruct)
+	//to add multiple certs with alias
+
+	var certificateName string
+
+	jwtConfArray := []SecurityTypeJWT{}
+	securityInstance := &wso2v1alpha1.Security{}
+	var certificateSecret = &corev1.Secret{}
+	for secName, scopeList := range securityMap {
+		//retrieve security instances
+		errGetSec := r.client.Get(context.TODO(), types.NamespacedName{Name: secName, Namespace: userNameSpace}, securityInstance)
+		if errGetSec != nil && errors.IsNotFound(errGetSec) {
+			log.Info("defined security instance " + secName + " is not found")
+			return securityDefinition, existSecCert, certList, jobVolumeMount, jobVolume, jwtConfArray, errGetSec
```

Comment:
The length is exceeded. Shall split into new line?

Reply:
Check other places too.

Reply:
Resolved in the latest code